src/algorithms/KirlikSayin.jl

```diff
@@ -0,0 +1,241 @@
+#  Copyright 2019, Oscar Dowson and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public License,
+#  v.2.0. If a copy of the MPL was not distributed with this file, You can
+#  obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+    KirlikSayin()
+
+`KirlikSayin` implements the Kirlik & Sayın (2014) algorithm.
+
+## Supported optimizer attributes
+
+"""
+
+mutable struct KirlikSayin <: AbstractAlgorithm
+  #yI::Union{Nothing, Vector{Float64}}
+  #yN::Union{Nothing, Vector{Float64}}
+  #L::Vector{Rectangle}
+
+  #KirlikSayin() = new(nothing, nothing, Rectangle[])
+  KirlikSayin() = new()
+end
+
+#=
+MOI.supports(::KirlikSayin, ::IdealPoint) = true
```

Comment:
For now I've abused `MOI.ObjectiveBound` to represent the ideal point: https://github.com/odow/MOO.jl/blob/78f47daa513c9f2aa3a21050174a2d2068a9e424/src/MOO.jl#L344-L355